public/components/agents/vuls/inventory.tsx

```diff
@@ -0,0 +1,218 @@
+  }
+
+  componentDidUpdate(prevProps) {
+    if (JSON.stringify(this.props.agent) !== JSON.stringify(prevProps.agent)){
+      this.setState({isLoading: true}, this.loadAgent)
+    }
+  }
+
+  componentWillUnmount() {
+    this._isMount = false;
+  }
+
+  async loadAgent() {
+    const {totalItems, items} = await this.getItemNumber();
+    if (this._isMount){
+      this.setState({ totalItems, items, isLoading: false });
+    }
+  }
+
+  getStoreFilters(props) {
+    const { section, selectView, agent } = props;
+    const filters = JSON.parse(window.localStorage.getItem(`wazuh-${section}-${selectView}-vulnerability-${agent['id']}`) || '{}');
+    return filters;
+  }
+
+  setStoreFilters(filters) {
+    const { section, selectView, agent } = this.props;
+    window.localStorage.setItem(`wazuh-${section}-${selectView}-vulnerability-${agent['id']}`, JSON.stringify(filters))
+  }
+
+  onFiltersChange = (filters) => {
+    this.setStoreFilters(filters);
+    this.setState({ filters });
+  }
+
+  onTotalItemsChange = (totalItems: number) => {
+    this.setState({ totalItemsFile: totalItems });
+  }
+
+  buildFilter() {
+    const filters = filtersToObject(this.state.filters);
+    const filter = {
+      ...filters,
+      limit: '15' 
+    };
+    return filter;
+  }
+
+  async getItemNumber() {
+    try {
+      const agentID = this.props.agent.id;
+      let response = await WzRequest.apiReq('GET', `/vulnerability/${agentID}`, {
+        params: this.buildFilter(),
+      });
+      return {
+        totalItems: ((response.data || {}).data || {}).total_affected_items || 0,
+        items: ((response.data || {}).data || {}).affected_items || [],
+      };
+    } catch (error) {
+      this.setState({ isLoading: false });
+      this.showToast('danger', error, 3000);
+    }
+  }
+
+  renderTabs() {
+    const { isLoading, totalItems } = this.state;
+    
+      return (
+        <EuiFlexGroup>
+          <EuiFlexItem>
+            <EuiTitle size="s">
+              <h1>Vulnerabilities&nbsp;{isLoading === true ? <EuiLoadingSpinner size="s" /> : <span>({ totalItems })</span>}</h1>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty iconType="importAction" onClick={() => this.downloadCsv()}>
+              Export formatted
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      )
+  }
+
+  showToast = (color, title, time) => {
+    getToasts().add({
+      color: color,
+      title: title,
+      toastLifeTimeMs: time,
+    });
+  };
+
+  async downloadCsv() {
+    const { filters } = this.state;
+    try {
+      const filtersObject = filtersToObject(filters);
+      const formatedFilters = Object.keys(filtersObject).map(key => ({name: key, value: filtersObject[key]}));
+      this.showToast('success', 'Your download should begin automatically...', 3000);
+      await exportCsv(
+        '/vulnerability/' + this.props.agent.id,
+        [{},
```

Comment:
Why this empty object?

Reply:
My bad, I forgot to deleted. I'll fix it right away